### PR TITLE
Vendor github.com/square/go-jose for kubeadm and kube-discovery

### PR DESCRIPTION
**Pull request by errordeveloper**
**What this PR does / why we need it**:

This is to be used by `kubeadm` (#31221) and @dgoodwin's `kube-discovery` (PR pending).

**_[xref kubernetes/features#11]**_

**_[cc @kubernetes/sig-cluster-lifecycle]**_

**Special notes for your reviewer**:

This is first time I've used `godep`, so please verify if all is quite right about the diff. I'm not 100% sure if there should be secondary dependencies or not.

``` release-note
NONE
```

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32190)

### CODE REVIEW

**Comment by thockin** [comment]
LGTM from a license and godeps POV - verify-godeps failed

**Comment by mikedanese** [comment]
I don't think this is going to pass verify until there is cod that's actually using it.

**Comment by luxas** [comment]
@errordeveloper Can you add some dummy code to `pkg/kubeadm/temp.go` that's using `go-jose` which you'll remove then?

In order to get this in... So reviewing becomes easier

**Comment by luxas** [comment]
Ping @errordeveloper Can you add the dummy code to this so we can get this in?
The gates are open for v1.5 things now...

**Comment by mikedanese** [comment]
Why don't we wait until there is code that actually uses this?

**Comment by luxas** [comment]
Since the kubeadm and kube-discovery PRs will be SO much easier to review if we get this in separately...

**Comment by errordeveloper** [comment]
I've added dummy code in `cmd/kubeadm` and `cmd/kube-discovery`, so whichever PR goes in first (@dgoodwin's or mine), will have to remove that.

**Comment by luxas** [comment]
Thanks @errordeveloper!
Still some verification errors though...
The kubeadm PR should remov the kubeadm dummy code and kube-discovery should remove the kube-discovery dummy code.

I'll add LGTM as soon as tests are passing

**Comment by luxas** [comment]
@errordeveloper Can you prioritize this so I can apply the LGTM label?

**Comment by errordeveloper** [comment]
Sure, will look into it first thing tomorrow.

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
 	"Packages": [
 		"github.com/ugorji/go/codec/codecgen",
 		"github.com/onsi/ginkgo/ginkgo",
+		"github.com/square/go-jose",
```
Comment (luxas) [inline Godeps/Godeps.json]: I think verify will pass if you remove this line

Comment (errordeveloper) [inline Godeps/Godeps.json]: Ah, I've added this before adding the dummy code to force it in. I'm still waiting for it to report back on latests changes, will try removing it, if it comes back with a failure.

Comment (errordeveloper) [inline Godeps/Godeps.json]: Ok, I can see the build log. This is clearly the problem, thanks.

Comment (errordeveloper) [inline Godeps/Godeps.json]: I've removed this line, rebased and squashed.

**Comment by errordeveloper** [comment]
Ok. I've updated `hack/.linted_packages` and believe this should be green shortly.

**Comment by luxas** [comment]
Thanks. LGTM
Readding the label

**Comment by luxas** [comment]
@errordeveloper Suddenly verify-golint failed, can you try to fix it?

**Comment by luxas** [comment]
Oh, ignore my last comment, web cache was stale

**Comment by eparis** [comment]
I'm dropping to p2. I don't know why this needs a priority label at all, but I want plenty of room above to get 1.4 items to the top of the queue.

**Pull request merged**